File: Backend/src/main/java/org/abrar/ecommerce/service/Cart/CartService.java
package org.abrar.ecommerce.service.Cart;

import org.abrar.ecommerce.dto.CartDTO;
import org.abrar.ecommerce.dto.CartItemDTO;

import java.util.List;

public interface CartService {
    CartDTO getCart(String email);

    CartDTO manageCartItem(String email, Long variantId, Integer quantity);

    CartDTO clearCart(String email);

    List<CartItemDTO> getCartItems(String email);
}

File: Frontend/src/api/cartService.js
import axiosInstance from './axiosInstance'

export const cartService = {
  getCart: async () => {
    const response = await axiosInstance.get('/cart')
    return response.data
  },

  getCartItems: async () => {
    const response = await axiosInstance.get('/cart/items')
    return response.data
  },

  manageItem: async (variantId, quantity) => {
    const response = await axiosInstance.post('/cart/manage', null, {
      params: { variantId, quantity },
    })
    return response.data
  },

  clearCart: async () => {
    const response = await axiosInstance.delete('/cart/clear')
    return response.data
  },
}